#!/usr/bin/env python
PACKAGE = "cameras_tryphon"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()


gen.add("cam1_gamma",    int_t,    0, "Camera 1: Gamma", 1,  0, 1)
gen.add("cam1_brightness",    int_t,    0, "Camera 1: Brightness", 240,  1, 255)
gen.add("cam1_shutter", double_t, 0, "Camera 1: Shutter",    16, 0.1,   33)
gen.add("cam1_gain", double_t, 0, "Camera 1: Gain",  0.4 , 0.4,   12)
gen.add("cam1_reverse", bool_t, 0, "Camera 1: reverse",  False)


gen.add("cam2_gamma",    int_t,    0, "Camera 2: Gamma", 1,  0, 1)
gen.add("cam2_brightness",    int_t,    0, "Camera 2: Brightness", 240,  1, 255)
gen.add("cam2_shutter", double_t, 0, "Camera 2: Shutter",    16, 0.1,   33)
gen.add("cam2_gain", double_t, 0, "Camera 2:  Gain",  .4 , 0.4,   12)
gen.add("cam2_reverse", bool_t, 0, "Camera 2: reverse",  False)


exit(gen.generate(PACKAGE, "cameras_tryphon", "pgrff_node_software"))
